Source/Core/Core/IOS/IOS.h

```diff
@@ -84,7 +84,10 @@ enum ProcessId : u32
 class Kernel
 {
 public:
+  // Used to initialise an IOS instance that is tied to PPC emulation.
   explicit Kernel(u64 ios_title_id);
+  // For internal uses. This only initialises core modules (ES, FS) and skips everything else.
+  Kernel();
```

Comment:
You're right, the GetDeviceByName is prone to abuse and by itself, a bit too error prone. I have changed Kernel to only expose ES and FS -- which are always available. This also removes the need to cast devices from users, which is (imo) a bit cleaner.